config/azure/publishRelease.py

```diff
@@ -76,18 +77,44 @@
 
 args = parser.parse_args()
 
-if not args.githubAccessToken :
-	parser.exit( 1, "No --github-access-token/GITHUB_ACCESS_TOKEN set")
-
 if not os.path.exists( args.archive ) :
 	parser.exit( 1, "The specified archive '%s' does not exist." % args.archive )
 
+# Validate the release contains our mandatory components
+
+requiredPaths = [
+	os.path.join( "doc", "gaffer", "html", "index.html" ),
+	os.path.join( "resources", "examples" )
+]
+
+for module in (
+	"Gaffer", "GafferAppleseed", "GafferArnold", "GafferDelight",
+	"GafferDispatch", "GafferImage", "GafferOSL", "GafferScene",
+	"GafferTractor", "GafferVDB"
+) :
+	requiredPaths.append( os.path.join( "python", module ) )
+	requiredPaths.append( os.path.join( "python", "%sUI" % module ) )
+
+with tarfile.open( args.archive, "r:gz" ) as a:
+	# getmember still reads the whole archive, so might as well grab them all
+	# as we go. We need to strip the first directory from all paths too as that
+	# contains the release name
+	archivePaths = [ os.path.join( *m.name.split( os.sep )[1:] ) for m in a.getmembers() if os.sep in m.name ]
```

Comment:
`{}` rather than `[]`, so the `in archivePaths` below isn't a linear search?

Reply:
Ahh thanks, I always for get you can do `set`s with `{}` too. https://github.com/GafferHQ/gaffer/commit/5d60155860378b80043b3de580cfa35b670534e8